plugins/unicodelib/unicodelib.cpp

```diff
@@ -719,6 +720,34 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+void translate(UnicodeString & source, UChar const * sear, unsigned searLen, UChar const * repl, unsigned replLen)
+{
+    UnicodeString search(sear, searLen);
+    UnicodeString replace(repl, replLen);
+    if (search.countChar32() != replace.countChar32() || source.isEmpty() || search.isEmpty() || replace.isEmpty())
+    {
+        return;
+    }
+    StringCharacterIterator it(source);
+    source.remove();
+    int32_t idx = it.setToStart();
+    while (idx != it.endIndex())
+    {
+        int32_t x = search.lastIndexOf(it.current32());
```

Comment:
When I tried using it.next32PostInc() I ran into issues with the code exiting on the last character without actually running it through because that member returns DONE if the next character is the last character. I'll try to work around it. I think I ran into that problem because I was using the member inside the while loop's condition.